packages/graphql/src/enums.ts

```diff
@@ -1137,3 +1137,17 @@ export enum SimpleCollectValidationFailedReason {
   NotEnoughBalance = 'NOT_ENOUGH_BALANCE',
   NotAFollower = 'NOT_A_FOLLOWER',
 }
+
+export enum FrameVerifySignatureResult {
```

Comment:
Can you please bind this new enum to the corresponding type in [here](https://github.com/lens-protocol/lens-sdk/blob/next/packages/graphql/src/graphql.ts#L111-L219)? This results in GQL inferred types that reference this enum.

Reply:
Added the binding

After I added it though [this line](https://github.com/defispartan/lens-sdk/blob/37e5f1e63c673003f5bcf128b488f6ad9bfb2795/packages/client/src/actions/frames.ts#L101) now returns a type error

Is there an example of another API endpoint that directly returns an enum? I tried searching to see how the types were implemented in the SDK but I couldn't find one

Reply:
Returning an enum should not be different than returning a scalar.

The closest thing I can think of is `Query. appServerApiKey` which returns a string.

Since you told the type system that the `FrameVerifySignatureResult` binds to the provided enum, you should not need to do lookup (with type assertion) at all.

```diff
export function verifyFrameSignature(
  client: AnyClient,
  request: VerifyFrameSignatureRequest,
): ResultAsync<FrameVerifySignatureResult, UnexpectedError> {
  return client
    .query(VerifyFrameSignatureQuery, { request })
-   .map((result) => FrameVerifySignatureResult[result as keyof typeof FrameVerifySignatureResult]);
}
```

Reply:
That worked, thanks!